examples/hosts/app-integration/document-recovery/src/recoveryAgent.ts

```diff
@@ -0,0 +1,145 @@
+        private readonly orgContainerId: string,
+        private readonly orgContainerSchema: ContainerSchema,
+    ) {
+        super();
+        const clientProps = {
+            connection: connectionConfig,
+        };
+        this.azureClient = new AzureClient(clientProps);
+        this.recoveredContainerId = undefined;
+        this.recoveryLog = "";
+        this.recoveryStatus = "NotStarted";
+    }
+
+    public static createRecoveryAgent(
+        containerId: string,
+        containerSchema: ContainerSchema,
+    ): RecoveryAgent {
+        return new RecoveryAgent(containerId, containerSchema);
+    }
+
+    public get getRecoveryInfo(): RecoveryInfo {
+        if (!this.orgContainerId) {
+            throw new Error("Cannot retreive status: Missing container ID.");
+        }
+
+        return {
+            originalContainerId: this.orgContainerId,
+            recoveryStatus: this.recoveryStatus,
+            recoveryLog: this.recoveryLog,
+            recoveredContainerId: this.recoveredContainerId,
+        };
+    }
+
+    public async startRecovery(): Promise<void> {
+        Promise.race([
+            this.recoverDoc(),
+            new Promise((_resolve, reject) =>
+                setTimeout(() => reject(new Error("timeout")), 1000),
+            ),
+        ]).then((val) => {
+            this.recoveredContainerId = val as string;
+            this.setRecoveryStatus(
+                "Success",
+                "Document succesfully recreated.",
+            );
+        }).catch((err) => {
+            this.setRecoveryStatus(
+                "Error",
+                err,
+            );
+        });
+    }
+
+    // Waiting on PR #9729  to expose getContainerVersions and recreateContainerFromVersion
+    private async recoverDoc(): Promise<string> {
+        this.setRecoveryStatus("KickedOff", "Kicked off recovery.");
+
+        /* Collect doc versions */
+        let versions: AzureContainerVersion[] = [];
+        try {
+            versions = await this.azureClient.getContainerVersions(
+                this.orgContainerId,
+                {
+                    maxCount: this.maxDocVersions,
+                },
+            );
+        } catch (e) {
+            return Promise.reject(new Error("Unable to get container versions."));
+        }
+
+        for (const version of versions) {
+            /* Attempt to copy doc from next available version of the older doc */
+            try {
+                const { container: newContainer } =
+                    await this.azureClient.copyContainer(
+                        this.orgContainerId,
+                        this.orgContainerSchema,
+                        version,
+                    );
+                const id = await newContainer.attach();
+
+                // FF Data may be lazy loaded on views. Fetch it now to ensure it's
+                // not currupted.
+                const sharedMap = newContainer.initialObjects.dataMap as SharedMap;
```

Comment:
So the core concept of this example is that we fetch older saved versions of this container, try to load it, then check each of it's fields and make sure they're in the same format as we expect. Is that a correct understanding?